building/build-debs/homeworld-keysystem/src/util/netutil/parseip.go

```diff
@@ -0,0 +1,45 @@
+package netutil
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+// returns IP address
+func ParseRemoteAddress(remote_addr string) (net.IP, error) {
+	index := strings.LastIndexByte(remote_addr, ':')
+	if index == -1 {
+		return nil, errors.New("Invalid request address (no colon)")
+	}
+	_, err := strconv.ParseUint(remote_addr[index+1:], 10, 16)
+	if err != nil {
+		return nil, err
+	}
+	var ip net.IP
+	if remote_addr[0] == '[' && remote_addr[index-1] == ']' {
+		ip = net.ParseIP(remote_addr[1 : index-1])
+		if ip == nil {
+			return nil, fmt.Errorf("Invalid request address (invalid IP in '%s')", remote_addr)
+		}
+		if ip.To4() != nil {
+			return nil, fmt.Errorf("IP address was expected to be IPv6, but was IPv4")
```

Comment:
What about, for instance, `[::ffff:0808:0808]` which is the IPv6-mapped version of the IPv4 address `8.8.8.8`?

Reply:
Hmm, yeah.